lib/chef/resource/chef_client_config.rb

```diff
@@ -0,0 +1,265 @@
+        coerce: proc { |x| x.map { |v| string_to_symbol(v).capitalize } },
+        default: []
+
+      # @todo we need to fixup bad plugin naming inputs here
+      property :ohai_optional_plugins, Array,
+        description: "Optional Ohai plugins that should be enabled to provide additional Ohai data for use in cookbooks.",
+        coerce: proc { |x| x.map { |v| string_to_symbol(v).capitalize } },
+        default: []
+
+      property :minimal_ohai, [true, false],
+        description: "Run a minimal set of Ohai plugins providing data necessary for the execution of #{ChefUtils::Dist::Infra::PRODUCT}'s built-in resources. Setting this to true will skip many large and time consuming data sets such as `cloud` or `packages`. Setting this this to true may break cookbooks that assume all Ohai data will be present."
+
+      property :start_handlers, Array,
+        description: "And an array of start handler hashes containing `class` and `argument` values.",
+        default: []
+
+      property :report_handlers, Array,
+        description: "And an array of report handler hashes containing `class` and `argument` values.",
+        default: []
+
+      property :exception_handlers, Array,
+        description: "And an array of exception handler hashes containing `class` and `argument` values.",
+        default: []
+
+      property :chef_license, String,
+        description: "Accept the [Chef EULA](https://www.chef.io/end-user-license-agreement/)",
+        equal_to: %w{accept accept-no-persist accept-silent}
+
+      property :policy_name, String,
+        description: "The name of a policy, as identified by the `name` setting in a Policyfile.rb file. `policy_group`  when setting this property."
+
+      property :policy_group, String,
+        description: "The name of a `policy group` that exists on the #{ChefUtils::Dist::Server::PRODUCT}. `policy_name` must also be specified when setting this property."
+
+      property :named_run_list, String,
+        description: "A specific named runlist defined in the node's applied Policyfile, which the should be used when running #{ChefUtils::Dist::Infra::PRODUCT}."
+
+      property :pid_file, String,
+        description: "The location in which a process identification number (pid) is saved. An executable, when started as a daemon, writes the pid to the specified file. "
+
+      property :file_cache_path, String,
+        description: "The location in which cookbooks (and other transient data) files are stored when they are synchronized. This value can also be used in recipes to download files with the `remote_file` resource."
+
+      property :file_backup_path, String,
+        description: "The location in which backup files are stored. If this value is empty, backup files are stored in the directory of the target file"
+
+      property :run_path, String,
+        description: "The location where #{ChefUtils::Dist::Infra::PRODUCT} should write the PID file."
```

Comment:
Is this description accurate? How does this differ from `pid_file`? I don't see `run_path` in the client.rb docs: https://docs.chef.io/config_rb_client/

Reply:
I was wondering about this one last night actually. I'll see why we historically had both